fast64_internal/oot/oot_operators.py

```diff
@@ -158,7 +170,29 @@ def execute(self, context):
 		csObj.select_set(True)
 		context.view_layer.objects.active = csObj
 		return {"FINISHED"}
-	
+
+class OOT_AddActor(bpy.types.Operator):
+	bl_idname = 'object.oot_add_actor'
+	bl_label = "Add Actor"
+	bl_options = {'REGISTER', 'UNDO', 'PRESET'}
+
+	scale : bpy.props.FloatProperty(default = 2)
+
+	def execute(self, context):
+		if context.mode != "OBJECT":
+			bpy.ops.object.mode_set(mode = "OBJECT")
+		bpy.ops.object.select_all(action = "DESELECT")
+
+		location = mathutils.Vector(bpy.context.scene.cursor.location)
+		bpy.ops.object.empty_add(type='CUBE', radius = 1, align='WORLD', location=location[:])
```

Comment:
a nice quality of life feature would be to add a property to the operator for setting the room the actor is parented to. it would show up in the redo panel. just a suggestion though

Reply:
I'm not sure what you mean there, like automatically parenting the actor to the room if there's one? if so, it could give issues if you have more than 1 room

Reply:
> like automatically parenting the actor to the room if there's one?

yes

> it could give issues if you have more than 1 room

I'm thinking the operator would have a room picker. (in the redo panel, not as a popup, and default would be to not parent to any room)

again this is just a feature suggestion, nothing mandatory to change

Reply:
I'll let this to someone else, I don't feel like doing this right now